prow/gerrit/client/client.go

```diff
@@ -193,10 +207,11 @@ func (c *Client) Start(cookiefilePath string) {
 
 // QueryChanges queries for all changes from all projects after lastUpdate time
 // returns an instance:changes map
-func (c *Client) QueryChanges(lastUpdate time.Time, rateLimit int) map[string][]ChangeInfo {
+func (c *Client) QueryChanges(lastState *LastSyncState, rateLimit int) map[string][]ChangeInfo {
```

Comment:
pointers to maps are awkward and often unnecessary. Why are you sure we need to do it this way?

Reply:
because we can't mutate maps passed by value? https://play.golang.org/p/gaGSuw4pvbm 
(not to say that we are doing it right now, but potentially could. better to support it since there's no harm either)

Reply:
You can: https://play.golang.org/p/b-nOoLSVFUp

Pass maps by value: https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

Reply:
yes if you allocate the memory for a map before hand, the map "value" is a reference to runime.hmap https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it
But like I pointed out, if we pass around nil maps then it doesn't work. 
That being said, we aren't leveraging this "feature", so lmk if you feel strongly about not using pointers.

Reply:
Yes, the awkward use of `(*lastState)[h.instance]` is clearly demonstrating that this has the wrong type.

Reply:
done